### PR TITLE
PR: Change optionsMap to a an array of tuples to be able to localize the options

### PR DESCRIPTION
**Pull request by goanpeca**
## Code changes

`NotebookTools.KeySelector.IOptions`

The `optionsMap` used in the `KeySelector` of the `notebooktools` has been renamed and restructured as an array of tuples, instead of an object.

These options are now localizable:

image

## User-facing changes

The export as options on the `File` menu has been shortened to avoid redundancy. The text in the palette has changed from `Export Notebook to {something}` to `Export Notebook: {something}`, which makes translations work as it is not a full sentence but two parts that are translated independently. (that is why we need to make a harder sentence split with the `:`)

### After
image

image

## Backwards-incompatible changes

OptionsMap that use the localizable string as the key. The names of these var will be changed to break (and force) the correct usage.

`KeySelector.IOptions.title` is no longer optional. We should encourage devs to give displayable/localizable names.

`KeySelector.IOptions.optionsMap` -> `KeySelector.IOptions.optionValueArray`

### CODE REVIEW

File: packages/notebook/src/notebooktools.ts [hunk 1]
```diff
@@ -38,6 +38,17 @@ import { NotebookPanel } from './panel';
...
+ * A type alias for a readonly partial JSON tuples `[option, value]`.
+ * `option` should be localized.
+ *
+ * Note: Partial here means that JSON object attributes can be `undefined`.
+ */
+export declare type ReadonlyPartialJSONOptionValueArray = [
+  ReadonlyPartialJSONValue | undefined,
+  ReadonlyPartialJSONValue
+][];
+
```
Comment (goanpeca) [inline packages/notebook/src/notebooktools.ts:51]: Where should this go @blink1073?

Also `export type` instead of  `export declare type` ?

Comment (blink1073) [reply packages/notebook/src/notebooktools.ts:51]: It should be exported as `export type` in the `NotebookTools` namespace.

File: packages/notebook-extension/src/index.ts [hunk 2]
```diff
@@ -469,18 +450,32 @@ function activateNotebookTools(
   optionsMap.None = null;
   void services.nbconvert.getExportFormats().then(response => {
     if (response) {
+      /**
+       * The exluded Cell Inspector Raw NbConvert Formats
```
Comment (blink1073) [inline packages/notebook-extension/src/index.ts:454]: excluded*

File: packages/notebook-extension/src/index.ts [hunk 3]
```diff
@@ -469,18 +450,32 @@ function activateNotebookTools(
...
+       * (returned from nbconvert's export list)
+       */
+      const rawFormtExclude = ['pdf', 'slides', 'script', 'notebook', 'custom'];
```
Comment (blink1073) [inline packages/notebook-extension/src/index.ts:457]: rawFormatExclude*

File: packages/notebook-extension/src/index.ts [hunk 4]
```diff
@@ -2089,6 +2084,23 @@ function populatePalette(
   });
 }
 
+/**
+ * The default Export To ... formats and their human readable labels.
+ */
+function getFormatLabels(translator?: ITranslator): { [k: string]: string } {
```
Comment (blink1073) [inline packages/notebook-extension/src/index.ts:2090]: Should be in the Private namespace

Comment (goanpeca) [reply packages/notebook-extension/src/index.ts:2090]: Fixed.

File: packages/notebook-extension/src/index.ts (hunk 3 above)
Comment (goanpeca) [reply packages/notebook-extension/src/index.ts:457]: Fixed

File: packages/notebook-extension/src/index.ts (hunk 2 above)
Comment (goanpeca) [reply packages/notebook-extension/src/index.ts:454]: Fixed

File: packages/notebook/src/notebooktools.ts (hunk 1 above)
Comment (goanpeca) [reply packages/notebook/src/notebooktools.ts]: Fixed!

**Pull request merged**